### PR TITLE
Create favorites tool

### PR DESCRIPTION
**Pull request by Kirollos-Hanna**

### CODE REVIEW

File: src/nova-components/CreateFavorites/resources/js/components/DropdownInput.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "DropdownInput",
+  beforeCreate: function () {
+    Nova.request()
+      .get("http://localhost:8000/nova-api/users")
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownInput.vue]: you should not add the app url, just /nova-api/users

File: src/nova-components/CreateFavorites/resources/js/components/DropdownInput.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    </div>
+    <div class="py-6 px-8 w-1/2">
+      <div class="flex items-center">
+        <select
+          data-testid="users-select"
+          dusk="user"
+          class="form-control form-select w-full"
+          :class="{ 'border-danger': error }"
+          v-bind:value="input"
+          v-on:input="$emit('input', $event.target.value)"
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownInput.vue]: don't use events to pass data to parent. hint: pass a function to the component as a prop

File: src/nova-components/CreateFavorites/resources/js/components/TextInput.vue [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      <div class="py-6 px-8 w-1/2">
+        <input
+          dusk="title"
+          list="title-list"
+          type="text"
+          :placeholder="placeholder"
+          class="w-full form-control form-input form-input-bordered"
+          :class="{ 'border-danger': error }"
+          v-bind:value="input"
+          v-on:input="$emit('input', $event.target.value)"
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/TextInput.vue]: don't use events to pass data to parent. hint: pass a function to the component as a prop

File: src/nova-components/CreateFavorites/resources/js/components/TextInput.vue [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<template>
+  <div class="card">
+    <div class="flex border-b border-40">
+      <div class="w-1/5 px-8 py-6">
+        <label for="title" class="inline-block text-80 pt-2 leading-tight">
+          {{ label }}
+          <span class="text-danger text-sm">*</span></label
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/TextInput.vue]: should not appear if input is not required.

File: src/nova-components/CreateFavorites/resources/js/components/Tool.vue [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      inputTitle: "",
+      inputSource: "",
+      inputUser: "",
+      errorTitle: false,
+      errorSource: false,
+      errorUser: false,
+    };
+  },
+  methods: {
+    submitForm: function (event) {
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/Tool.vue:95]: can be written in a better way?

File: src/nova-components/CreateFavorites/resources/js/components/Tool.vue [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+      formData.append("user", this.inputUser);
+      formData.append("user_trashed", false);
+      formData.append("viaResource", "");
+      formData.append("viaResourceId", "");
+      formData.append("viaRelationship", "");
+
+      Nova.request()
+        .post("/nova-api/favorites?editing=true&editMode=create", formData)
+        .then((res) => {
+          window.location.href = "/nova";
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/Tool.vue]: use vue-router for navigation

File: src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<template>
+  <a href="/nova" class="btn btn-link dim cursor-pointer text-80 ml-auto mr-6">
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue]: Use your own style
+ Component name says "button", but you are using a link?

File: src/nova-components/CreateFavorites/apollo.config.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+const path = require('path')
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/apollo.config.js]: why are you using apollo?

File: src/app/Nova/User.php [hunk 9]
```diff
@@ -48,23 +48,23 @@ public function fields(Request $request)
 
             Gravatar::make()->maxWidth(50),
 
-            Text::make('Name', 'name')
+            Text::make('Name')
```
Comment (Mhany95) [inline src/app/Nova/User.php]: Why did you remove the names?

File: src/nova-components/CreateFavorites/resources/js/components/DateField.vue [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  </div>
+</template>
+
+<script>
+import Label from "./Label.vue";
+
+export default {
+  name: "DateField",
+  props: {
+    label: String,
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/DateField.vue]: add default values to all of your props

File: src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+</template>
+
+<script>
+import Label from "./Label.vue";
+
+export default {
+  name: "DropdownField",
+  beforeCreate: function () {
+    Nova.request()
+      .get("http://localhost:8000/nova-api/users")
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue]: incorrect url

File: src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+      .then((res) => {
+        let resources = res.data.resources;
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue]: create a helper function in a helper file that parses the response from nova

File: src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue [hunk 13]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <a href="#" class="cancel" @click="redirect"> Cancel </a>
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue]: use a button not a link

File: src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  methods: {
+    redirect: function () {
+      this.$router.push("/dashboards/main");
+    },
+  },
+};
+</script>
+
+<style>
+.cancel {
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue]: Put style in a separate file

File: src/nova-components/CreateFavorites/resources/js/components/DateField.vue [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+  </div>
+</template>
+
+<script>
+import Label from "./Label.vue";
+
+export default {
+  name: "DateField",
+  props: {
+    label: "",
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/DateField.vue]: props should have default values and types

File: src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue [hunk 16]
```diff
@@ -0,0 +1,56 @@
+<template>
+  <div class="field-container">
+    <div class="label-spacing">
+      <Label :label="label" :isRequired="true" />
+    </div>
+    <div class="input-spacing">
+      <select
+        id="users-select"
+        class="input-decoration arrow-decoration"
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue]: class is defined twice and an Id is already given to the component

File: src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+        :class="{ 'input-error': error }"
+        v-model="input"
+        @change="$emit('changeInput', $data.input, label)"
+      >
+        <option value="" selected="selected" disabled="disabled">—</option>
+        <option v-for="user in users" :value="user.id" :key="user.name">
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue:15]: Pass id as a key

File: src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+      .catch((e) => console.log(e));
+  },
+  data() {
+    return {
+      users: [],
+      input: "",
+    };
+  },
+  props: {
+    label: "",
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue]: props should have default values and types

File: src/nova-components/CreateFavorites/resources/js/components/Label.vue [hunk 19]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <label :for="label">
+    {{ label }}
+    <span v-if="isRequired" class="text-danger text-sm">*</span></label
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/Label.vue]: text-sm and text-danger are not your own style

File: src/nova-components/CreateFavorites/resources/js/components/Label.vue [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+  >
+</template>
+
+<script>
+export default {
+  name: "Label",
+  props: {
+    label: "",
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/Label.vue]: props should have default values and types

File: src/nova-components/CreateFavorites/resources/js/components/SubmitButton.vue [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+</template>
+
+<script>
+export default {
+  name: "SubmitButton",
+};
+</script>
+
+<style>
+.submit-button {
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/SubmitButton.vue]: Put style in a separate file

File: src/nova-components/CreateFavorites/resources/js/components/TextField.vue [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+</template>
+
+<script>
+import Label from "./Label.vue";
+import TextInput from "./TextInput.vue";
+
+export default {
+  name: "TextField",
+  props: {
+    label: "",
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/TextField.vue]: props should have default values and types

File: src/nova-components/CreateFavorites/resources/js/components/TextField.vue [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+    type: "",
+    error: false,
+    validationError: false,
+    isRequired: false,
+  },
+  components: {
+    Label,
+    TextInput,
+  },
+  data: function () {
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/TextField.vue]: Unify how you define your functions across your project

File: src/nova-components/CreateFavorites/resources/js/components/TextInput.vue [hunk 24]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <input
+    :type="type"
+    class="input-decoration"
+    :class="{ 'input-error': error }"
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/TextInput.vue]: class is already defined

File: src/nova-components/CreateFavorites/resources/js/components/TextInput.vue [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+
+export default {
+  name: "TextInput",
+  data: function () {
+    return {
+      input: "",
+    };
+  },
+  props: {
+    placeholder: "",
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/TextInput.vue]: props should have default values and types

File: src/nova-components/CreateFavorites/resources/js/components/Tool.vue [hunk 26]
```diff
@@ -0,0 +1,263 @@
...
+        this.errorUser = false;
+      } else if (type === this.postDate) {
+        this.inputPostDate = input;
+        this.errorPostDate = false;
+      } else if (type === this.author) {
+        this.inputAuthor = input;
+      }
+    },
+    validateUrl(urlString) {
+      var pattern = new RegExp(
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/Tool.vue]: Put logic in a helper function to be reused in other components if needed

File: src/nova-components/CreateFavorites/resources/js/components/Tool.vue [hunk 27]
```diff
@@ -0,0 +1,263 @@
...
+      this.errorSource = this.inputSource === "" ? true : false;
+      this.errorUser = this.inputUser === "" ? true : false;
+      this.errorPostDate = this.inputPostDate === "" ? true : false;
+    },
+  },
+};
+</script>
+
+<style>
+h1 {
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/components/Tool.vue]: Put style in a separate file

File: src/nova-components/CreateFavorites/resources/js/helpers.js [hunk 28]
```diff
@@ -0,0 +1,9 @@
+export function parseUserResponse(response) {
```
Comment (Mhany95) [inline src/nova-components/CreateFavorites/resources/js/helpers.js]: This method is specifically parsing Ids and names only, we want it to parse every possible field that a response might have, so don't treat this as something specific for users

File: src/composer.json [hunk 29]
```diff
@@ -60,6 +62,14 @@
...
             "url": "./nova"
+        },
+        {
+            "type": "path",
+            "url": "./nova-components/CreateFavorites"
+        },
+        {
+            "type": "path",
+            "url": "./nova-components/Test"
         }
```
Comment (A-Elshorafa) [inline src/composer.json]: Remove Old component reference

File: src/database/migrations/2020_12_08_160641_update_favorites_table.php [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        //
+    }
```
Comment (A-Elshorafa) [inline src/database/migrations/2020_12_08_160641_update_favorites_table.php]: Add dropIfExist()

File: src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue [hunk 31]
```diff
@@ -0,0 +1,12 @@
+<template>
+  <button class="cancel">Cancel</button>
+</template>
+
+<script>
+export default {
+  name: "CancelButton",
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/CancelButton.vue]: Component name should be in kabab-case

File: src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue [hunk 32]
```diff
@@ -0,0 +1,73 @@
...
+    </div>
+    <div class="input-spacing">
+      <select
+        :class="{
+          'input-error': error,
+          'input-decoration': true,
+          'arrow-decoration': true,
+        }"
+        v-model="input"
+        @change="$emit('changeInput', $data.input, label)"
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/DropdownField.vue]: don't use events pass function as a prop to component then call this function inside component

File: src/nova-components/CreateFavorites/resources/js/components/TextField.vue [hunk 33]
```diff
@@ -0,0 +1,89 @@
...
+    return {
+      input: "",
+    };
+  },
+  methods: {
+    changeInputField: function (...args) {
+      const [input] = args;
+      this.input = input;
+      this.$emit("changeInput", input, this.$props.label);
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/TextField.vue]: don't use events pass function as a prop to component then call this function inside component

File: src/nova-components/CreateFavorites/resources/js/components/TextInput.vue [hunk 34]
```diff
@@ -0,0 +1,63 @@
+<template>
+  <input
+    :type="type"
+    :class="{
+      'input-error': error || validationError || sameError,
+      'input-decoration': true,
+    }"
+    :placeholder="placeholder"
+    v-model="input"
+    @input="$emit('changeInputField', $data.input)"
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/TextInput.vue]: don't use events pass function as a prop to component then call this function inside component

File: src/nova-components/CreateFavorites/resources/js/components/TextField.vue [hunk 35]
```diff
@@ -0,0 +1,89 @@
...
+    },
+    setInput: function () {
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/TextField.vue]: not used function

File: src/nova-components/CreateFavorites/resources/js/components/TextInput.vue [hunk 36]
```diff
@@ -0,0 +1,63 @@
...
+    sameError: {
+      default: false,
+      type: Boolean,
+    },
+  },
+  components: {
+    Label,
+  },
+  methods: {
+    setInput: function (input) {
```
Comment (abdelrahmanna) [inline src/nova-components/CreateFavorites/resources/js/components/TextInput.vue:55]: not used function

File: src/nova-components/CreateFavorites/resources/js/components/TextField.vue (hunk 35 above)
Comment (Kirollos-Hanna) [reply src/nova-components/CreateFavorites/resources/js/components/TextField.vue]: It's called from the parent component

File: src/nova-components/CreateFavorites/resources/js/components/TextInput.vue (hunk 36 above)
Comment (Kirollos-Hanna) [reply src/nova-components/CreateFavorites/resources/js/components/TextInput.vue:55]: It's called from the parent component

File: src/database/migrations/2020_12_08_160641_update_favorites_table.php [hunk 37]
```diff
@@ -26,6 +26,6 @@ public function up()
      */
     public function down()
     {
-        //
+        Schema::dropIfExists('favorites');
```
Comment (abdelrahmanna) [inline src/database/migrations/2020_12_08_160641_update_favorites_table.php]: drop columns that was added in this migration only

File: src/nova-components/CreateFavorites/.env.local [hunk 38]
```diff
@@ -0,0 +1 @@
+VUE_APP_APOLLO_ENGINE_KEY=y
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/.env.local]: Why you create `.env.local`

File: src/nova-components/CreateFavorites/resources/js/components/Tool.vue [hunk 39]
```diff
@@ -0,0 +1,189 @@
+<template>
+  <div id="content">
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/Tool.vue]: shouldn't to use `id` property

File: src/nova-components/CreateFavorites/resources/js/components/inputs/TextInput.vue [hunk 40]
```diff
@@ -0,0 +1,65 @@
...
+</template>
+
+<script>
+export default {
+  name: "text-input",
+  data: function () {
+    return {
+      input: "",
+    };
+  },
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/inputs/TextInput.vue]: Make `input` as computed, And make it updated with props

File: src/nova-components/CreateFavorites/resources/js/components/inputs/TextInput.vue [hunk 41]
```diff
@@ -0,0 +1,65 @@
...
+        return;
+      },
+      type: Function,
+    },
+  },
+  methods: {
+    setInput: function (input) {
+      this.input = input;
+    },
+  },
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/inputs/TextInput.vue]: No need to this method when add computed for `input`

File: src/nova-components/CreateFavorites/resources/js/components/Tool.vue [hunk 42]
```diff
@@ -0,0 +1,189 @@
...
+      this.inputUser = "";
+      this.inputPostDate = "";
+      this.inputAuthor = "";
+    },
+    removeInputValues: function () {
+      this.$refs.userField.setInput("");
+      this.$refs.titleField.setInput("");
+      this.$refs.sourceField.setInput("");
+      this.$refs.postDateField.setInput("");
+      this.$refs.authorField.setInput("");
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/Tool.vue]: No need to `$refs` when you add `computed()` in TextInput.vue

File: src/nova-components/CreateFavorites/resources/js/components/Tool.vue [hunk 43]
```diff
@@ -0,0 +1,189 @@
+<template>
+  <div class="content">
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/Tool.vue]: There's no class called `content` in tool.css, Remove undefined class names please

File: src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue [hunk 44]
```diff
@@ -0,0 +1,80 @@
...
+        resources.map((user) => {
+          this.users.push({ name: user.title, id: user.id.value });
+        });
+      })
+      .catch((e) => console.log(e));
+  },
+  data() {
+    return {
+      users: [],
+      input: "",
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue]: Make input as `computed()`

File: src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue [hunk 45]
```diff
@@ -0,0 +1,80 @@
...
+    },
+  },
+  components: {
+    Label,
+  },
+  methods: {
+    setInput: function (input) {
+      this.input = input;
+    },
+  },
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue]: No need `setInput()` when use computed

File: src/nova-components/CreateFavorites/resources/js/components/labels/index.js [hunk 46]
```diff
@@ -0,0 +1,4 @@
+import Label from "./Label.vue";
+
+
+export {Label};
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/labels/index.js]: Components should follow kabab-case eg: 'label-component'

File: src/nova-components/CreateFavorites/resources/js/components/fields/TextField.vue [hunk 47]
```diff
@@ -0,0 +1,96 @@
...
+      <p v-if="error && validationError" class="error">
+        The {{ label }} field is required and must be a Url.
+      </p>
+      <p v-else-if="error" class="error">The {{ label }} field is required.</p>
+      <p v-else-if="validationError" class="error">
+        The {{ label }} field must be a Url.
+      </p>
+      <p v-else-if="sameError" class="error">
+        This {{ label }} is already used.
+      </p>
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/fields/TextField.vue]: Create component for error `<p>` eg: `<error-labal ...>`, And replace it by `<p class="error">`

File: src/nova-components/CreateFavorites/resources/js/components/fields/TextField.vue [hunk 48]
```diff
@@ -0,0 +1,96 @@
...
+    };
+  },
+  methods: {
+    changeInputField: function (...args) {
+      const [input] = args;
+      this.input = input;
+      this.$emit("changeInput", input, this.$props.label);
+    },
+    setInput: function () {
+      this.$refs.input.setInput("");
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/fields/TextField.vue]: No need to `this.$refs` when use `computed()`

File: src/nova-components/CreateFavorites/resources/js/components/fields/TextField.vue [hunk 49]
```diff
@@ -0,0 +1,96 @@
...
+  data: function () {
+    return {
+      input: "",
+    };
+  },
+  methods: {
+    changeInputField: function (...args) {
+      const [input] = args;
+      this.input = input;
+      this.$emit("changeInput", input, this.$props.label);
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/fields/TextField.vue]: Use `changeInput` as a method by calling `this.changeInput(...)`,Please don't miss to make validation on it's type before use it (should typeof === 'function')

File: src/nova-components/CreateFavorites/resources/js/components/inputs/TextInput.vue [hunk 50]
```diff
@@ -0,0 +1,65 @@
+<template>
+  <input
+    :type="type"
+    :class="{
+      'input-error': error || validationError || sameError,
+      'input-decoration': true,
+    }"
+    :placeholder="placeholder"
+    v-model="input"
+    @input="changeInput($data.input, placeholder)"
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/inputs/TextInput.vue]: Add a method and call `this.changeInput()` in it, Then mention it with `@input`

File: src/nova-components/CreateFavorites/resources/js/components/inputs/textInput.css [hunk 51]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+
+.input-error {
+    border-color: red;
+}
+  
+.disabled-input {
+    background-color: #eee;
+}
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/inputs/textInput.css]: Remove unused classes

File: src/nova-components/CreateFavorites/resources/js/components/labels/Label.vue [hunk 52]
```diff
@@ -0,0 +1,32 @@
...
+  props: {
+    label: {
+      default: {
+        default: "",
+        type: String,
+      },
+      type: {
+        default: false,
+        type: Boolean,
+      },
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/labels/Label.vue]: Remove Unused code

File: src/nova-components/CreateFavorites/resources/js/components/labels/Label.vue [hunk 53]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    isRequired: {
+      default: false,
+      type: Boolean,
+    },
+  },
+};
+</script>
+
+<style scoped>
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/labels/Label.vue]: What is the usage of `scoped` here?

File: src/nova-components/CreateFavorites/resources/js/components/tool.css [hunk 54]
```diff
@@ -0,0 +1,35 @@
...
+    -webkit-appearance: none;
+    appearance: none;
+    background-image: linear-gradient(45deg, transparent 50%, gray 50%),
+      linear-gradient(135deg, gray 50%, transparent 50%),
+      radial-gradient(#ddd 70%, transparent 72%);
+    background-position: calc(100% - 20px) calc(1em + 2px),
+      calc(100% - 15px) calc(1em + 2px), calc(100% - 0.5em) 0.5em;
+    background-size: 5px 5px, 5px 5px, 1.5em 1.5em;
+    background-repeat: no-repeat;
+  }
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/tool.css]: Remove unused class

File: src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue [hunk 55]
```diff
@@ -0,0 +1,80 @@
...
+    Nova.request()
+      .get("/nova-api/users")
+      .then((res) => {
+        const resources = parseResponse(res);
+
+        resources.map((user) => {
+          this.users.push({ name: user.title, id: user.id.value });
+        });
+      })
+      .catch((e) => console.log(e));
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue]: Remove Console.log(), Also remove logs from any files

File: src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue [hunk 56]
```diff
@@ -0,0 +1,80 @@
...
+  name: "dropdown-field",
+  beforeCreate: function () {
+    Nova.request()
+      .get("/nova-api/users")
+      .then((res) => {
+        const resources = parseResponse(res);
+
+        resources.map((user) => {
+          this.users.push({ name: user.title, id: user.id.value });
+        });
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/components/fields/DropdownField.vue]: `.map()` should return your the needed values, Please use it properly.

File: src/nova-components/CreateFavorites/resources/js/helpers.js [hunk 57]
```diff
@@ -0,0 +1,20 @@
+export function parseResponse(response) {
+    return response.data.resources;
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/helpers.js]: Please validate `response`, `response.data` and `response.data.resources`

File: src/nova-components/CreateFavorites/resources/js/helpers.js [hunk 58]
```diff
@@ -0,0 +1,20 @@
...
+        "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // port and path
+        "(\\?[;&a-z\\d%_.~+=-]*)?" + // query string
+        "(\\#[-a-z\\d_]*)?$",
+      "i"
+    ); // fragment locator
+    return !!pattern.test(urlString);
+}
+
+export function validateEmptyInput (input) {
+    return input === "" ? true : false;
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/helpers.js]: No need to `true : false` just leave the logic return it's result

File: src/nova-components/CreateFavorites/resources/js/tool.js [hunk 59]
```diff
@@ -0,0 +1,9 @@
+Nova.booting((Vue, router, store) => {
+  router.addRoutes([
+    {
+      name: 'create-favorites',
+      path: '/create-favorites',
+      component: require('./components/Tool'),
+    },
+  ])
+})
```
Comment (A-Elshorafa) [inline src/nova-components/CreateFavorites/resources/js/tool.js]: Use `Vue` to define your components check `Vue.component()`

File: src/routes/api.php [hunk 60]
```diff
@@ -29,6 +29,6 @@
 //     return response()->json(['Laravel 7 CORS Demo']);
 // });
 
-// Route::middleware('auth:api')->get('/user', function (Request $request) {
-//     return $request->user();
-// });
+Route::middleware('auth:api')->get('/user', function (Request $request) {
+    return $request->user();
+});
```
Comment (A-Elshorafa) [inline src/routes/api.php]: 1-Make `UserController.php` for future work
2-Add this function to user controller and link this route to controller function 
3-Remove commented code in this file
